minimal-plugin/src/main/java/io/micronaut/gradle/graalvm/NativeLambdaExtension.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2003-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gradle.graalvm;
+
+import org.gradle.api.provider.Property;
+
+/**
+ * Allows configuring the native lambda specific parameters.
+ */
+public interface NativeLambdaExtension {
+    /**
+     * The native lambda runtime
+     * @return the runtime
+     */
+    Property<NativeLambdaRuntime> getLambdaRuntime();
```

Comment:
Would it be possible for this to be a String instead of an enum? That way it would be more extensible and flexible. Users may be able to write other main classes. 

We will generate in starter the appropriate  String.

Reply:
I think we can support both. I don't like stringly typed APIs much when we can do better. Especially here the user would have to figure out the main classes. Even if you help in starter, switching from one to the other would be more complicated.

What about having :

```
public interface NativeLambdaExtension {
    /**
     * The native lambda runtime
     * @return the runtime
     */
    Property<NativeLambdaRuntime> getLambdaRuntime();

    Property<String> getLambdaRuntimeClassName();
}
```

where the 2d property would let the user specify an explicit class name, but the default value would be the enum class?

Reply:
I pushed the changes I was thinking about, let me know